src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -245,10 +246,22 @@ def __call__(self, offset, length, exists):
         return total_used_size, snap_map
 
     @classmethod
-    def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
+    def _rbd_image(cls, ioctx, pool_name, namespace, image_name):  # pylint: disable=R0912
         with rbd.Image(ioctx, image_name) as img:
-
             stat = img.stat()
+            snapshot_mode = img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_SNAPSHOT
+            if img.mirror_image_get_mode() == rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL:
+                stat['mirror_mode'] = 'journal'
+            elif snapshot_mode:
```

Comment:
If we don't want to invoke multiple times a foreign API (because the overhead might be non negligible), then let's store the mirror mode instead:
```suggestion
            mirror_mode = img.mirror_image_get_mode()
            if mirror_mode == rbd.RBD_MIRROR_IMAGE_MODE_JOURNAL:
                stat['mirror_mode'] = 'journal'
            elif mirror_mode == rbd.RBD_MIRROR_IMAGE_MODE_SNAPSHOT:
```